TopologyChangeTest.when_jobParticipantReceivesStaleInitOperation_then_jobRestarts[2]

```
Error Message
Values should be different. Actual: -7533742546029853916
Stacktrace
java.lang.AssertionError: Values should be different. Actual: -7533742546029853916
	at com.hazelcast.jet.core.TopologyChangeTest.when_jobParticipantReceivesStaleInitOperation_then_jobRestarts(TopologyChangeTest.java:393)
```

https://hazelcast-l337.ci.cloudbees.com/job/Jet-pr-builder/com.hazelcast.jet$hazelcast-jet-core/2559/testReport/junit/com.hazelcast.jet.core/TopologyChangeTest/when_jobParticipantReceivesStaleInitOperation_then_jobRestarts_2_/

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/TopologyChangeTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/TopologyChangeTest.java
@@ -34,6 +34,7 @@ import com.hazelcast.jet.impl.MasterContext;
 import com.hazelcast.jet.impl.execution.init.JetInitDataSerializerHook;
 import com.hazelcast.jet.impl.operation.InitExecutionOperation;
 import com.hazelcast.test.HazelcastParametersRunnerFactory;
+import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
@@ -380,6 +381,16 @@ public class TopologyChangeTest extends JetTestSupport {
 
         // When
         long executionId = masterContext.getExecutionId();
+
+        assertTrueEventually(() -> {
+            Arrays.stream(instances)
+                  .filter(instance -> !instance.getHazelcastInstance().getCluster().getLocalMember().isLiteMember())
+                  .filter(instance ->  instance != instances[2])
+                  .map(JetTestSupport::getJetService)
+                  .map(service -> service.getJobExecutionService().getExecutionContext(executionId))
+                  .forEach(Assert::assertNotNull);
+        });
+
         newInstance.getHazelcastInstance().getLifecycleService().terminate();
         for (JetInstance instance : instances) {
             assertClusterSizeEventually(NODE_COUNT, instance.getHazelcastInstance());

